client/play-client/src/main/scala/sttp/tapir/client/play/EndpointToPlayClient.scala

```diff
@@ -95,15 +94,49 @@ private[play] class EndpointToPlayClient(clientOptions: PlayClientOptions, ws: S
           case EndpointIO.FixedHeader(_, codec, _)         => codec.decode(())
           case EndpointIO.Empty(codec, _)                  => codec.decode(())
           case EndpointOutput.OneOf(mappings, codec) =>
-            mappings
-              .find(mapping => mapping.statusCode.isEmpty || mapping.statusCode.contains(code)) match {
-              case Some(mapping) =>
-                getOutputParams(mapping.output, body, headers, code, statusText).flatMap(p => codec.decode(p.asAny))
+            val mappingsForStatus = mappings collect {
+              case m if m.statusCode.isEmpty || m.statusCode.contains(code) => m
+            }
+
+            val contentType = headers
+              .get(HeaderNames.ContentType)
+              .map { h => MediaType.parse(h.head) }
+
+            def applyMapping(m: EndpointOutput.StatusMapping[_]) =
+              getOutputParams(m.output, body, headers, code, statusText).flatMap(p => codec.decode(p.asAny))
+
+            contentType match {
               case None =>
-                DecodeResult.Error(
-                  statusText,
-                  new IllegalArgumentException(s"Cannot find mapping for status code ${code} in outputs $output")
-                )
+                val firstNonBodyMapping = mappingsForStatus.find(_.output.traverseOutputs {
+                  case _ @(EndpointIO.Body(_, _, _) | EndpointIO.StreamBodyWrapper(_)) => Vector(false)
+                  case _                                                               => Vector(true)
```

Comment:
this won't recurse - see the traverse outputs impl. I think matching on `_: EndpointOuput.Basic` should do the trick here

Reply:
Or better: 
```
_.output.traverseOutputs {
  case Body => Vector(())
}.isEmpty
```